hbase-client/src/main/java/org/apache/hadoop/hbase/MetaTableAccessor.java

```diff
@@ -1580,6 +1580,7 @@ public static void splitRegion(Connection connection, RegionInfo parent, long pa
       Put putParent = makePutFromRegionInfo(RegionInfoBuilder.newBuilder(parent)
                         .setOffline(true)
                         .setSplit(true).build(), time);
+      addRegionStateToPut(putParent, State.SPLIT);
```

Comment:
I think for split, it is SPLIT_PARENT? And what is the way we do this for branch-1? In the above code, the RegionInfo has been set to offline and also split, so I do not think it should come back online again? If it does then there are some bugs in other places.

Reply:
The SPLIT_PARENT add back when enable table, and the code is in: 
https://github.com/apache/hbase/blob/master/hbase-server/src/main/java/org/apache/hadoop/hbase/master/procedure/EnableTableProcedure.java#L114
https://github.com/apache/hbase/blob/master/hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/RegionStates.java#L329
https://github.com/apache/hbase/blob/master/hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/RegionStates.java#L349
And i not am familiar with all this logic.